src/FCharacter.cpp

```diff
@@ -142,6 +142,11 @@ void FCharacter::fire()
 	m_weapons[m_currentWeapon].fire(/*m_facingLeft ? FDirection::left : FDirection::right*/);
 }
 
+void FCharacter::reload(float delta)
+{
+	m_weapons[m_currentWeapon].tick(delta);
+}
```

Comment:
This is a great solution to making it frame-invariant, *but* a more clean way of doing this is to make `FCharacter::tick()` call `tick()` on all weapons in `m_weapons`. That way all weapons are reloaded, not just the current one. (Maybe or may be not we only want to load the current weapon, that would create very different gameplay, but let's see later)